include/boost/astronomy/coordinate/base_frame.hpp

```diff
@@ -13,89 +13,91 @@
 #include <boost/astronomy/coordinate/representation.hpp>
 #include <boost/astronomy/coordinate/differential.hpp>
 #include <boost/astronomy/detail/is_base_template_of.hpp>
+#include <boost/astronomy/coordinate/arithmetic.hpp>
 
 
-namespace boost 
+namespace boost { namespace astronomy { namespace coordinate {
+
+typedef boost::geometry::degree degree;
+typedef boost::geometry::radian radian;
+
+namespace bu = boost::units;
+namespace bg = boost::geometry;
+
+template <typename Representation, typename Differential>
+struct base_frame
 {
-    namespace astronomy
+    ///@cond INTERNAL
+    BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+        <boost::astronomy::coordinate::base_representation, Representation>::value),
+        "First template argument is expected to be a representation class");
+    BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+        <boost::astronomy::coordinate::base_differential, Differential>::value),
+        "second template argument is expected to be a differential class");
+    ///@endcond
+
+protected:
+    Representation data;
+    Differential motion;
+
+public:
+
+    //!returns 2d propermotion
+    auto proper_motion() const
+    {
+        return std::make_tuple
+        (
+            bg::get<0>(motion.get_differential()) *
+                typename Differential::quantity1::unit_type(),
```

Comment:
Instead of `unit_type` syntax you should use `quantity::from_value` syntax as once you pointed out it gives error.

Reply:
Yes, I was trying to remember that I have pointed some error with one of the methods but couldn't remember anything so I moved on with this method. I'll change this now.